ui.js

```diff
@@ -0,0 +1,135 @@
+'use strict';
+const dns = require('dns');
+const React = require('react');
+const {useState, useEffect} = require('react');
+const Spinner = require('ink-spinner').default;
+const {Box, Text, Newline, useApp} = require('ink');
+
+const api = require('./api');
+
+const FixedSpacer = ({size}) => (
+	<>{' '.repeat(size)}</>
+);
+
+const ErrorMessage = ({text}) => (
+	<Box>
+		<Text bold color="red">
+			›
+			<FixedSpacer size={1}/>
+		</Text>
+		<Text dimColor>
+			{text}
+		</Text>
+		<Newline count={2}/>
+	</Box>
+);
+
+const Spacer = ({singleLine}) => {
+	if (singleLine) {
+		return null;
+	}
+
+	return (
+		<Text>
+			<Newline count={1}/>
+		</Text>
+	);
+};
+
+const DownloadSpeed = ({isDone, downloadSpeed, uploadSpeed, downloadUnit} = {}) => {
+	const color = (isDone || uploadSpeed) ? 'green' : 'cyan';
+
+	return (
+		<Text color={color}>
+			{downloadSpeed}
+			<FixedSpacer size={1}/>
+			<Text dimColor>{downloadUnit}</Text>
+			<FixedSpacer size={1}/>
+			↓
+		</Text>
+	);
+};
+
+const UploadSpeed = ({isDone, uploadSpeed, uploadUnit} = {}) => {
+	const color = isDone ? 'green' : 'cyan';
+
+	if (uploadSpeed) {
+		return (
+			<Text color={color}>
+				{uploadSpeed}
+				<Text dimColor>
+					{` ${uploadUnit} ↑`}
+				</Text>
+			</Text>
+		);
+	}
+
+	return <Text dimColor color={color}>{' - Mbps ↑'}</Text>;
+};
+
+const Speed = ({upload, data}) => upload ? (
+	<>
+		<DownloadSpeed {...data}/>
+		<Text dimColor>{' / '}</Text>
+		<UploadSpeed {...data}/>
+	</>
+) : (<DownloadSpeed {...data}/>);
+
+const Fast = ({singleLine, upload}) => {
+	const [error, setError] = useState('');
+	const [data, setData] = useState({});
+	const [isDone, setIsDone] = useState(false);
+	const {exit} = useApp();
+
+	useEffect(() => {
+		dns.lookup('fast.com', error => {
+			if (error && error.code === 'ENOTFOUND') {
+				setError('Please check your internet connection');
+				exit();
+			} else {
```

Comment:
```suggestion
			if (error && error.code === 'ENOTFOUND') {
				setError('Please check your internet connection');
				exit();
				return;
			}
```

to reduce nesting.

Reply:
And shouldn't it still show the error when the `error.code` is not `ENOTFOUND`?

Reply:
Reduced nesting and added displaying errors for other cases